lint/linter.py

```diff
@@ -1298,18 +1298,22 @@ def run(self, cmd, code):
 
     def get_tempfile_suffix(self):
         """Return the mapped tempfile_suffix."""
-        if self.tempfile_suffix and not self.view.file_name():
-            if isinstance(self.tempfile_suffix, dict):
-                suffix = self.tempfile_suffix.get(util.get_syntax(self.view), self.syntax)
-            else:
-                suffix = self.tempfile_suffix
+        suffix = ''
 
-            if not suffix.startswith('.'):
-                suffix = '.' + suffix
+        if isinstance(self.tempfile_suffix, str):
+            suffix = self.tempfile_suffix
+        elif isinstance(self.tempfile_suffix, dict):
+            suffix = self.tempfile_suffix.get(util.get_syntax(self.view), self.syntax)
 
-            return suffix
-        else:
-            return ''
+        # default to plain text
```

Comment:
I don't think we *need* an extension. Just leave it empty if none was specified. 
Alternatively, you could try inferring it from the view's file name, if that exists.

Reply:
Is this the logic that you are suggesting for that method?

1. Extract the extension from `self.view.file_name()` and return if available
2. Infer the extension from the currently selected syntax and return if possible
3. Use `self.tempfile_suffix` in any other cases
4. Never return empty

Please confirm so I can commit these changes.